libs/barista-components/breadcrumbs/src/breadcrumbs.ts

```diff
@@ -49,38 +98,394 @@ export const _DtBreadcrumbMixinBase = mixinColor<
+          if (
+            event.target === firstTransplantedItem?._elementRef.nativeElement
+          ) {
+            event.preventDefault();
+            this._trigger?.nativeElement.focus();
+            return;
+          }
+
+          // Focus the last item in the overlay if the last visible is blurred and the overlay is open
+          if (
+            event.target === firstVisibleItem?._elementRef.nativeElement &&
+            lastTransplantedItem
+          ) {
+            event.preventDefault();
+            lastTransplantedItem._elementRef.nativeElement.focus();
+            return;
+          }
+
+          return;
+        }
+
+        // Focus first focusable item in the overlay
+        if (
+          event.target === this._trigger?.nativeElement &&
+          this._overlayRef?.hasAttached() &&
+          firstTransplantedItem
+        ) {
+          event.preventDefault();
+          firstTransplantedItem._elementRef.nativeElement.focus();
+          return;
+        }
+
+        // Focus the next focusable item if the last item in the overlay is blurred
+        if (event.target === lastTransplantedItem?._elementRef.nativeElement) {
+          if (firstVisibleItem) {
+            event.preventDefault();
+            firstVisibleItem._elementRef.nativeElement.focus();
+          } else {
+            this._trigger?.nativeElement.focus();
+          }
+
+          return;
+        }
+      }
+    });
+  }
+
+  /** Nexts overlay trigger observable to trigger focus handling */
+  _handleOverlayTriggerKeydown($event: KeyboardEvent): void {
+    this._overlayTriggerKeydown$.next($event);
+  }
+
+  /**
+   * Creates a container in the body that holds all breadcrumb items
+   * This container is then moved into the overlay when the overlay is created
+   */
+  private _createCollapsedContainer(): void {
+    if (!this._collapsedContainer) {
+      const container = this._document.createElement('div');
+      container.id = `dt-breadcrumb-collapsed-container`;
+      container.setAttribute(`aria-expanded`, `false`);
+      container.classList.add('dt-breadcrumb-collapsed-container');
+      this._document.body.appendChild(container);
+      this._collapsedContainer = container;
+    }
+  }
+
+  /**
+   * Returns an array of items that need to be moved to the overlay
+   * @param containerRect Bounds of the breadcrumbs container
+   */
+  private _determineItemsToTransplant(
+    containerRect: DOMRect,
+  ): DtBreadcrumbsItem2[] {
+    return determineOverflowingItems(
+      containerRect,
+      this._itemsWidthMap,
+      COLLAPSED_BUTTON_WIDTH,
+    );
+  }
+
+  /** Returns the first focusable item of the breadcrumbs in the overlay */
+  private _getFirstTransplantedFocusableItem(): DtBreadcrumbsItem2 | null {
+    const focusableTransplantedItems = Array.from(
+      this._itemsPortalsMap.keys(),
+    ).filter((item) => item._isFocusable);
+    return focusableTransplantedItems[0] || null;
```

Comment:
See the comments below for refactoring these. 
I'd suggest creating a findFirstFocusableItem pure function that takes an array of DtBreadcrumbsItem2 and returns the result of `find((item) => item._isFocusable)`.

You can then use it in the two methods below as well.